pkg/frontend/querymiddleware/stats.go

```diff
@@ -107,6 +121,17 @@ func (s queryStatsMiddleware) populateQueryDetails(ctx context.Context, req Requ
 	}
 }
 
+func (s queryStatsMiddleware) consistencyCounters(ctx context.Context) {
```

Comment:
[nit] Would you consider improving the `Test_queryStatsMiddleware_Do`? Looks easy to do to test the new metric too.

Reply:
added in https://github.com/grafana/mimir/pull/7091/commits/bad87f1a26e3261caf5f5d6cf7518dc57f176b33